chainio/clients/elcontracts/writer.go

```diff
@@ -442,3 +451,325 @@ func (w *ChainWriter) ProcessClaims(
 
 	return receipt, nil
 }
+
+func (w *ChainWriter) ForceDeregisterFromOperatorSets(
+	ctx context.Context,
+	operator gethcommon.Address,
+	avs gethcommon.Address,
+	operatorSetIds []uint32,
+	waitForReceipt bool,
+) (*gethtypes.Receipt, error) {
+	if w.avsDirectory == nil {
+		return nil, errors.New("AVSDirectory contract not provided")
+	}
+
+	noSendTxOpts, err := w.txMgr.GetNoSendTxOpts()
+	if err != nil {
+		return nil, utils.WrapError("failed to get no send tx opts", err)
+	}
+
+	tx, err := w.allocationManager.DeregisterFromOperatorSets(
+		noSendTxOpts,
+		allocationmanager.IAllocationManagerTypesDeregisterParams{
+			Operator:       operator,
+			Avs:            avs,
+			OperatorSetIds: operatorSetIds,
+		},
+	)
+
+	if err != nil {
+		return nil, utils.WrapError("failed to create ForceDeregisterFromOperatorSets tx", err)
+	}
+
+	receipt, err := w.txMgr.Send(ctx, tx, waitForReceipt)
+	if err != nil {
+		return nil, utils.WrapError("failed to send tx", err)
+	}
+
+	return receipt, nil
+}
+
+func (w *ChainWriter) ModifyAllocations(
+	ctx context.Context,
+	operatorAddress gethcommon.Address,
+	allocations []allocationmanager.IAllocationManagerTypesAllocateParams,
+	waitForReceipt bool,
+) (*gethtypes.Receipt, error) {
+	if w.allocationManager == nil {
+		return nil, errors.New("AllocationManager contract not provided")
+	}
+
+	noSendTxOpts, err := w.txMgr.GetNoSendTxOpts()
+	if err != nil {
+		return nil, utils.WrapError("failed to get no send tx opts", err)
+	}
+
+	tx, err := w.allocationManager.ModifyAllocations(noSendTxOpts, operatorAddress, allocations)
+	if err != nil {
+		return nil, utils.WrapError("failed to create ModifyAllocations tx", err)
+	}
+
+	receipt, err := w.txMgr.Send(ctx, tx, waitForReceipt)
+	if err != nil {
+		return nil, utils.WrapError("failed to send tx", err)
+	}
+
+	return receipt, nil
+}
+
+func (w *ChainWriter) SetAllocationDelay(
+	ctx context.Context,
+	operatorAddress gethcommon.Address,
+	delay uint32,
+	waitForReceipt bool,
+) (*gethtypes.Receipt, error) {
+	if w.allocationManager == nil {
+		return nil, errors.New("AllocationManager contract not provided")
+	}
+
+	noSendTxOpts, err := w.txMgr.GetNoSendTxOpts()
+	if err != nil {
+		return nil, utils.WrapError("failed to get no send tx opts", err)
+	}
+
+	tx, err := w.allocationManager.SetAllocationDelay(noSendTxOpts, operatorAddress, delay)
+	if err != nil {
+		return nil, utils.WrapError("failed to create InitializeAllocationDelay tx", err)
+	}
+	receipt, err := w.txMgr.Send(ctx, tx, waitForReceipt)
+	if err != nil {
+		return nil, utils.WrapError("failed to send tx", err)
+	}
+
+	return receipt, nil
+}
+
+func (w *ChainWriter) DeregisterFromOperatorSets(
```

Comment:
I think we still need to support the legacy registration and deregistration flows in the sdk see: https://github.com/Layr-Labs/eigenlayer-contracts/blob/813f52cd125b9fb46f69fb5679a930b030187a2b/src/contracts/core/AVSDirectory.sol#L71 as they'll be used to migrate iirc

Reply:
that method in is called by AVS service manager iirc right? I think we never had those in core bindings as they would never be called directly.

Reply:
I think we maintained it in the core bindings (at least we did with register as operator) bc not all avs were using our service manager

https://github.com/Layr-Labs/eigensdk-go/blob/2c554f1a02be8e59be2e712a35551e297700d89a/chainio/clients/elcontracts/writer.go#L157

Reply:
we still have that - https://github.com/Layr-Labs/eigensdk-go/blob/859288176f423c69637a21a6ef50fabf73f7475a/chainio/clients/elcontracts/writer.go#L167 
but this is for register to eigenlayer core, it has no relation with AVS registration per se.

Reply:
ahh ok my bad!